clang/test/CodeGenSYCL/loop_fusion_host.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -triple -x86_64-unknown-linux-gnu -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-host -triple -x86_64-unknown-linux-gnu -disable-llvm-passes -verify -Wno-sycl-2017-compat -DDIAG %s
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel(const Func &kernelFunc) {
+  kernelFunc();
+}
+
+template <int SIZE>
+class KernelFunctor5 {
+public:
+  [[intel::loop_fuse(SIZE)]] void operator()() const {}
+};
+
+template <int SIZE>
+class KernelFunctor3 {
+public:
+  [[intel::loop_fuse_independent(SIZE)]] void operator()() const {}
+};
+
+[[intel::loop_fuse]] void func1() {}
+[[intel::loop_fuse_independent]] void func2() {}
+
+void foo() {
+
+  KernelFunctor5<5> f5;
+  kernel<class kernel_name_1>(f5);
+
+  KernelFunctor5<3> f3;
+  kernel<class kernel_name_1>(f5);
+}
+// CHECK-NOT: !loop_fuse
+
+#if defined(DIAG)
+int baz();
+[[intel::loop_fuse(baz())]] void func3(); // expected-error{{'loop_fuse' attribute requires an integer constant}}
```

Comment:
Note, that this diagnostic will be emitted two times (or even 3 I guess, because integration header generation is additional front-end invocation), for host and device mode both in regular use case of SYCL compiler.

Reply:
Oh, that is rather unfortunate behavior, isn't it? :-/ But won't that be true for just about any warning in their code? e.g., if they have a signed/unsigned mismatch in their code, won't they get that warning 2-3 times as well?

Reply:
well, WARNINGS do, ERRORs do not.  The ERROR won't get hit 2-3x, since our driver only runs the 2nd/3rd compile if the 1st succeeds (and only the 3rd if the 2nd succeeds, etc).

Reply:
>>won't they get that warning 2-3 times as well
Yep. This is the reasoning I used (personally) to justify the multiple diagnostics.